runtime/jcl/common/java_lang_invoke_MethodHandleNatives.cpp

```diff
@@ -303,13 +309,14 @@ getSignatureFromMethodType(J9VMThread *currentThread, j9object_t typeObject, UDA
 		}
 	}
 
-	*signatureLength += strlen(rSignature);
+	signatureLength += strlen(rSignature);
 
-	methodDescriptor = (char*)j9mem_allocate_memory(*signatureLength+1, OMRMEM_CATEGORY_VM);
+	methodDescriptor = (J9UTF8*)j9mem_allocate_memory(signatureLength + sizeof(J9UTF8) + 1, OMRMEM_CATEGORY_VM);
```

Comment:
Why `+1`?  J9UTF8's aren't null terminated

Reply:
the `strcpy` used for copying rSignature appends an extra null terminator.
Should I change it to strncpy/memcpy ?

Reply:
Yes, it should be consistent.  We shouldn't sometimes null terminate j9utf8s

Reply:
changed strcpy, to strncpy